bitwarden_license/src/CommCore/Services/ProviderService.cs

```diff
@@ -362,9 +368,23 @@ public async Task<ProviderOrganization> CreateOrganizationAsync(Guid providerId,
             return providerOrganization;
         }
 
-        // TODO: Implement this
-        public Task RemoveOrganization(Guid providerOrganizationId, Guid removingUserId) => throw new NotImplementedException();
-        
+        public async Task RemoveOrganization(Guid providerId, Guid providerOrganizationId, Guid removingUserId)
+        {
+            var providerOrganization = await _providerOrganizationRepository.GetByIdAsync(providerOrganizationId);
+
+            if (providerOrganization == null || providerOrganization.ProviderId != providerId)
+            {
+                throw new BadRequestException("Invalid organization");
+            }
+            
+            if (!await _organizationService.HasConfirmedOwnersExceptAsync(providerOrganization.OrganizationId, new Guid[] {}))
```

Comment:
I don't see the current provider organization owner id being passed in to check before deleting, or I'm not understanding the intent of this check completely. Why check here if there's at least 1 confirmed owner when you're going to delete one, what if there's exactly only 1 left, wouldn't that cause an issue when deleting? Or, is this an org having its own owner vs. the provider owner? That would make sense, just not sure.

Reply:
It's a check to ensure the organization has an owner. Since it will efficiently be disconnected from the provider it could end up orphaned without any user (or owner).